libcudacxx/include/cuda/std/detail/libcxx/include/tuple

```diff
@@ -282,25 +356,20 @@ public:
                   "temporary whose lifetime has ended");
   }
 
-  template <class _Tp, class _Alloc>
-  _LIBCUDACXX_INLINE_VISIBILITY explicit __tuple_leaf(integral_constant<int, 1>, const _Alloc& __a, _Tp&& __t)
-      : __value_(allocator_arg_t(), __a, _CUDA_VSTD::forward<_Tp>(__t))
+  __tuple_leaf(const __tuple_leaf& __t) = default;
+  __tuple_leaf(__tuple_leaf&& __t)      = default;
+
+  _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX14 __tuple_leaf& operator=(const __tuple_leaf& __t) noexcept
   {
-    static_assert(!_CCCL_TRAIT(is_reference, _Hp),
-                  "Attempted to uses-allocator construct a reference element in a tuple");
+    __value_ = __t.__value_;
+    return *this;
   }
-
-  template <class _Tp, class _Alloc>
-  _LIBCUDACXX_INLINE_VISIBILITY explicit __tuple_leaf(integral_constant<int, 2>, const _Alloc& __a, _Tp&& __t)
-      : __value_(_CUDA_VSTD::forward<_Tp>(__t), __a)
+  _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX14 __tuple_leaf& operator=(__tuple_leaf&& __t) noexcept
   {
-    static_assert(!_CCCL_TRAIT(is_reference, _Hp),
-                  "Attempted to uses-allocator construct a reference element in a tuple");
+    __value_ = __t.__value_;
```

Comment:
```suggestion
    __value_ = _CUDA_VSTD::move(__t.__value_);
```

Reply:
I also saw this, but this is in the specialization where `__value_` is a reference. Doesn't moving do the wrong thing here?

Reply:
I am pretty sure that this is technically correct, especially given that this could also happen for any user defined type that has a move operator.

Also note that previously this was implemented by a template with a forwarding reference.

https://godbolt.org/z/5KnT1vd56